README.md

```diff
@@ -14,10 +14,38 @@ Curve fitting method in javascript
 
 ## [API Documentation](https://mljs.github.io/levenberg-marquardt/)
 
+This algorithm it's based on the article [Transtrum, Mark K., Benjamin B. Machta, and James P. Sethna. "Geometry of nonlinear least squares with applications to sloppy models and optimization." Physical Review E 83.3 (2011): 036701.](https://doi.org/10.1103/PhysRevE.83.036701)
+
+In order to get a general idea of the problem you could also check the [Wikipedia article](https://en.wikipedia.org/wiki/Levenberg%E2%80%93Marquardt_algorithm).
+
 ## Example
 
 ```js
-const levenbergMarquardt = require('ml-levenberg-marquardt');
+// import library
+const LM = require('ml-levenberg-marquardt');
+
+// function that receives the parameters and returns
+// a function with the independent variable as a parameter
+function sinFunction (a, b) {
+  return (t) => (a * Math.sin(b * t));
+}
+
+// array of points to fit
+let data = [ /* [x1, x2, ... ], [y1, y2, ... ] */ ];
+
+// array of initial parameter values
+let initialValues = [ /* a, b, c, ... */ ];
+
+const options = {
+  dampingIncrease: 10e-3,
+  dampingDecrease: 10e-3,
+  maxIterations: 100,
+  errorTolerance: 10e-3
+};
+
+let damping = 2;
+
+let fittedParams = LM(data, initialValues, damping, sinFunction, options);
```

Comment:
+1 for 3 parameters. I would put the function first though because it's the main thing this algorithm is about.
As for putting mandatory parameters in the options, I think it makes sense generally (when there are many parameters) because it's easier to remember the name of a parameter than its place in the arguments list